xcube/webapi/ows/coverages/controllers.py

```diff
@@ -77,62 +79,110 @@ def get_coverage_data(
+                f'{collection_id}: {", ".join(unrecognized_vars)}'
+            )
+
+    # https://docs.ogc.org/DRAFTS/19-087.html#datetime-parameter-subset-requirements
+    # requirement 7D: "If a datetime parameter is specified requesting a
+    # coverage without any temporal dimension, the parameter SHALL either be
+    # ignored, or a 4xx client error generated." We choose to ignore it.
+    if request.datetime is not None and 'time' in ds.variables:
+        if isinstance(request.datetime, tuple):
+            time_slice = slice(*request.datetime)
+            time_slice = ensure_time_index_compatible(ds, time_slice)
+            ds = ds.sel(time=time_slice)
         else:
+            timespec = ensure_time_index_compatible(ds, request.datetime)
             ds = ds.sel(time=timespec, method='nearest').squeeze()
-    if 'properties' in query:
-        # TODO: Raise an exception for non-existent properties
-        vars_to_keep = set(query['properties'][0].split(','))
-        data_vars = set(ds.data_vars)
-        vars_to_drop = list(data_vars - vars_to_keep)
-        ds = ds.drop_vars(vars_to_drop)
-    if 'crs' in query:
-        target_crs = query['crs'][0]
-        source_gm = GridMapping.from_dataset(ds)
-        target_gm = source_gm.transform(target_crs).to_regular()
+
+    if request.subset is not None:
+        subset_bbox, ds = _apply_subsetting(ds, request.subset, subset_crs)
+    else:
+        subset_bbox = None
+
+    if request.bbox is not None:
+        ds = _apply_bbox(ds, request.bbox, bbox_crs, always_xy=False)
+
+    _assert_coverage_size_ok(ds, request.scale_factor)
+
+    source_gm = GridMapping.from_dataset(ds, crs=native_crs)
+    target_gm = None
+
+    if native_crs != final_crs:
+        target_gm = source_gm.transform(final_crs).to_regular()
+    if request.scale_factor != 1:
+        if target_gm is None:
+            target_gm = source_gm
+        target_gm = target_gm.scale(1. / request.scale_factor)
+    if request.scale_axes is not None:
+        # TODO implement scale-axes
+        raise ApiError.NotImplemented(
+            'The scale-axes parameter is not yet supported.'
+        )
+    if request.scale_size:
+        # TODO implement scale-size
+        raise ApiError.NotImplemented(
+            'The scale-size parameter is not yet supported.'
+        )
+
+    if target_gm is not None:
         ds = resample_in_space(ds, source_gm=source_gm, target_gm=target_gm)
 
+    # In this case, the transformed native CRS bbox[es] may have been
+    # too big, so we re-crop in the final CRS.
+    if native_crs != final_crs and request.bbox is not None:
+        ds = _apply_bbox(ds, request.bbox, bbox_crs, always_xy=False)
+    if subset_bbox is not None:
+        ds = _apply_bbox(ds, subset_bbox, subset_crs, always_xy=True)
+
+    ds.rio.write_crs(final_crs, inplace=True)
+    for var in ds.data_vars.values():
+        var.attrs.pop('grid_mapping', None)
+
+    # TODO rename axes to match final CRS?
```

Comment:
Is this really a TODO? Otherwise

```suggestion
    # check: rename axes to match final CRS?
```